pkg/apis/configuration/v1/kongplugin_types.go

```diff
@@ -36,6 +36,7 @@ import (
 // +kubebuilder:printcolumn:name="Programmed",type=string,JSONPath=`.status.conditions[?(@.type=="Programmed")].status`
 // +kubebuilder:validation:XValidation:rule="(has(self.config) || has(self.configFrom)) ? ((!has(self.config) && has(self.configFrom)) || (has(self.config) && !has(self.configFrom))) : true", message="Using both config and configFrom fields is not allowed."
 // +kubebuilder:validation:XValidation:rule="self.plugin == oldSelf.plugin", message="The plugin field is immutable"
+// +kubebuilder:validation:XValidation:rule="(has(self.config)) || (!has(self.config) && !has(self.configPatches))", message="Can use 'configPatches' only when 'config' specified"
```

Comment:
Ditto question about the restriction on requiring config.

If we _do_ remove that we would want a rule that disallows setting both `configPatches` and `configFrom`, since `configPatches` would imply `config` even if `config` itself is empty.

Reply:
updated the CEL here.